### DISCUSSION TITLE
Allowed memory size of 1073741824 bytes exhausted (tried to allocate 356527440 bytes)

### DISCUSSION THREAD

**Discussion by ajaskiewiczpl** [firefly-iii/firefly-iii#7367] [Firefly III]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [ ] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely a Firefly III issue, not me](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

I am running Firefly 6.0.6 in Docker container on Raspberry Pi 4 (64 bit) and after upgrading from Firefly 5.x it started to show errors on web page after a while with below error message:
`Allowed memory size of 1073741824 bytes exhausted (tried to allocate 356527440 bytes) `
Restarting docker container helps, but it happens again after a few hours. My docker compose file:
```version: "3.3"

services:
  fireflyiii:
    external_links:
      - mariadb
    image: fireflyiii/core:latest
    container_name: firefly
    volumes:
      - /mnt/storage/firefly/upload:/var/www/html/storage/upload
    env_file: .env
    ports:
      - 8099:8080
    networks:
      - mariadb_shared
    restart: unless-stopped
networks:
  mariadb_shared:
    external: true
```

### Debug information

Debug information generated at 2023-04-12 00:00:00 Europe/Warsaw for Firefly III version **6.0.6**.

| Scope | Version |
| --- | --- |
| Firefly III | 6.0.6 |
| Firefly III API | 2.0.1 |
| Build | 441, 01-04-2023 08:04:21 CEST |
| DB version | 19 (exp. 19) |
| PHP | `8.2.4` |
| Host | `Linux` |

| System info | Value |
| --- | --- |
| System TZ | Europe/Warsaw |
| Browser TZ | Europe/Warsaw |
| App environment | production |
| App debug mode | false |
| App cache driver | file |
| App logging | info, stack |
| Display errors | Off |
| Error reporting | ALL errors |
| Max upload | 67108864 (64 MB) |
| Interface | apache2handler |
| Default language | en_US |
| Default locale | equal |
| BCscale | 12 |
| DB drivers | mysql, pgsql, sqlite |
| Current driver | mysql |
| Trusted proxies (.env) | `**` |

| User info | Value |
| --- | --- |
| Login provider | eloquent  |
| User guard | web  |
| Headers | N/A, N/A  |
| Stateful domains |  |
| Session start |  |
| Session end |  |
| Session first |  |
| User ID | 1 |
| User language | en_US |
| User locale | pl_PL |
| Attempt at "pl_PL.utf8" | 'pl_PL.utf8' |
| Attempt at "pl_PL.UTF-8" | 'pl_PL.UTF-8' |
| User agent | Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/111.0 |

**Comment by JC5** [comment]
What page are you trying to visit when this happens?

**Comment by ajaskiewiczpl** [comment]
The main page.
I forgot to mention that I am running docker on Raspberry Pi 4 64-bit (updated original post)

**Comment by JC5** [comment]
Could you tell me how much data you're trying to load?

**Comment by ajaskiewiczpl** [comment]
Not sure if I understand the question, but on main page I have transactions for current month only by default (30-70 transactions)

**Comment by JC5** [comment]
Alright. See if you can find which page crashes exactly.

https://docs.firefly-iii.org/firefly-iii/faq/other/#how-do-i-enable-debug-mode

**Comment by ajaskiewiczpl** [comment]
Ok I think we have it:
```
[Thu Apr 13 08:08:13.947616 2023] [php:error] [pid 246] [client 172.20.0.1:51594] PHP Fatal error:  Allowed memory size of 1073741824 bytes exhausted (tried to allocate 356527440 bytes) in /var/www/html/vendor/monolog/monolog/src/Monolog/Formatter/LineFormatter.php on line 113
```

**Comment by JC5** [comment]
What are the surrounding lines of log, please? I wonder what was happening.

**Comment by ajaskiewiczpl** [comment]
```
127.0.0.1 - - [14/Apr/2023:09:00:33 +0200] "GET /robots.txt HTTP/1.1" 200 237 "-" "Firefly III Health Checker/1.0"
127.0.0.1 - - [14/Apr/2023:09:01:33 +0200] "GET /robots.txt HTTP/1.1" 200 237 "-" "Firefly III Health Checker/1.0"
127.0.0.1 - - [14/Apr/2023:09:02:33 +0200] "GET /robots.txt HTTP/1.1" 200 237 "-" "Firefly III Health Checker/1.0"
[2023-04-14T09:02:37.145852+02:00] production.INFO: AUDIT: User visits homepage. (160.220.200.230 (redacted@gmail.com) -> GET:https://firefly.redacted.com) [] []
[Fri Apr 14 09:02:56.397345 2023] [php:error] [pid 233] [client 172.20.0.1:43882] PHP Fatal error:  Allowed memory size of 1073741824 bytes exhausted (tried to allocate 356527440 bytes) in /var/www/html/vendor/monolog/monolog/src/Monolog/Formatter/LineFormatter.php on line 113
172.20.0.1 - - [14/Apr/2023:09:02:36 +0200] "GET / HTTP/1.0" 500 4044 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/111.0"
172.20.0.1 - - [14/Apr/2023:09:02:56 +0200] "GET /v1/lib/bs/css/bootstrap.min.css?v=6.0.6 HTTP/1.0" 200 20062 "https://firefly.redacted.com/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/111.0"
172.20.0.1 - - [14/Apr/2023:09:02:56 +0200] "GET /v1/css/gf-source.css HTTP/1.0" 200 1181 "https://firefly.redacted.com/v1/lib/adminlte/css/AdminLTE.min.css?v=6.0.6" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/111.0"
172.20.0.1 - - [14/Apr/2023:09:02:56 +0200] "GET /v1/lib/adminlte/css/skins/skin-blue-light.min.css?v=6.0.6 HTTP/1.0" 404 3802 "https://firefly.redacted.com/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/111.0"
172.20.0.1 - - [14/Apr/2023:09:02:57 +0200] "GET /v1/fonts/source-sans-pro-v13-greek_cyrillic-ext_vietnamese_greek-ext_latin-ext_cyrillic_latin-regular.woff2 HTTP/1.0" 200 62955 "https://firefly.redacted.com/v1/css/gf-source.css" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/111.0"
127.0.0.1 - - [14/Apr/2023:09:03:34 +0200] "GET /robots.txt HTTP/1.1" 200 237 "-" "Firefly III Health Checker/1.0"
```
Before the error there are only those `Firefly III Health Checker/1.0` occurring for couple of hours.

**Comment by JC5** [comment]
Mmm. There's not much I can do about this. Try allocating more memory to the Docker container. Some processes are memory intensive.

**Comment by ajaskiewiczpl** [comment]
I think it's related to scheduled cron job. I configured host system to run cron on the docker container daily at 5 am. Here is the command:
`docker exec firefly /usr/local/bin/php /var/www/html/artisan firefly-iii:cron`
When I disable cron job, firefly is not crashing and is working fine.
